HLTrigger/Tools/python/convertToRaw.py

```diff
@@ -0,0 +1,134 @@
+    mergingPset = cms.untracked.string( "" ),
+    outputAdler32Recheck = cms.untracked.bool( False ),
+)
+
+process.writer = cms.OutputModule("RawStreamFileWriterForBU",
+    source = cms.InputTag('rawDataCollector'),
+    numEventsPerFile = cms.uint32(0)                                # to be overwritten after parsing the command line options
+)
+
+process.endpath = cms.EndPath(process.writer)
+
+process.load('FWCore.MessageService.MessageLogger_cfi')
+process.MessageLogger.cerr.FwkReport.reportEvery = 0                # to be overwritten after parsing the command line options
+
+# parse command line options
+options = VarParsing.VarParsing ('python')
+for name in 'filePrepend', 'maxEvents', 'outputFile', 'secondaryOutputFile', 'section', 'tag', 'storePrepend', 'totalSections':
+    del options._register[name]
+    del options._beenSet[name]
+    del options._info[name]
+    del options._types[name]
+    if name in options._singletons:
+        del options._singletons[name]
+    if name in options._lists:
+        del options._lists[name]
+    if name in options._noCommaSplit:
+        del options._noCommaSplit[name]
+    if name in options._noDefaultClear:
+        del options._noDefaultClear[name]
+
+
+options.register('runNumber',
+                 0,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.int,
+                 "Run number to use")
+
+options.register('lumiNumber',
+                 None,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.int,
+                 "Luminosity section number to use")
+
+options.register('eventsPerLumi',
+                 11650,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.int,
+                 "Number of events in the given luminosity section to process")
+
+options.register('eventsPerFile',
+                 50,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.int,
+                 "Split the output into files with at most this number of events")
+
+options.register('outputPath',
+                 os.getcwd(),
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.string,
+                 "Output directory for the FED RAW data files")
+
+options.parseArguments()
+
+# check that the option values are valide
+if options.runNumber == 0:
+    sys.stderr.write('Invalid run number\n')
+    sys.exit(1)
+
+if options.lumiNumber == 0:
+    sys.stderr.write('Invalid luminosity section number\n')
+    sys.exit(1)
+
+if options.eventsPerLumi == 0:
+    sys.stderr.write('Invalid number of events per luminosity section\n')
+    sys.exit(1)
+
+if options.eventsPerFile == 0:
+    sys.stderr.write('Invalid number of events per output file\n')
+    sys.exit(1)
```

Comment:
What about negative values for `runNumber` and the other args?

Out of curiosity: how do you decide between `sys.stderr + exit` and other types of exits? (naively, I would have used `raise Exception` and removed `import sys`)

Reply:
By the way, `eventsPerLumi == -1` is valid, and means "no limit".

Reply:
As for the exceptions, I prefer the error message I get from `sys.stderr`:
```sh
$ python3 HLTrigger/Tools/python/convertToRaw.py inputFiles=file.root runNumber=0
Invalid run number
```
to the message I get from exceptions:
```sh
$ python3 HLTrigger/Tools/python/convertToRaw.py inputFiles=file.root runNumber=0
Traceback (most recent call last):
  File "/fff/fwyzard/CMSSW_12_3_5/src/HLTrigger/Tools/python/convertToRaw.py", line 104, in <module>
    raise RuntimeError('Invalid run number')
RuntimeError: Invalid run number
```

As a user, why should I care about the traceback and where in the code the exception was generated ?

Reply:
In more complex cases, it might be useful to know where the code is breaking, but in this case it would indeed be obvious. One could do `raise SystemExit("error message")` (seems equivalent: same error message to stderr, exit code 1, less source code, no `import sys`), but it's a minor point of course.